File: Coding Sample/MS.NET/Fundamentals/Concurrency/threadtest.cs
using System;
using System.Threading;

static class Program
{
	private static void HandleJob(int id)
	{
		Console.WriteLine("Thread<{0}> has accepted job:{1}", Thread.CurrentThread.ManagedThreadId, id);
		Worker.DoWork(id);
		Console.WriteLine("Thread<{0}> has finished job:{1}", Thread.CurrentThread.ManagedThreadId, id);
	}

	public static void Main(string[] args)
	{
		int n = args.Length > 0 ? Convert.ToInt32(args[0]) : 50;

		Thread child = new Thread(delegate()
		{
			HandleJob(n);
		});
		child.IsBackground = n > 70;
		child.Start();

		HandleJob(40);
	}
}

File: Coding Sample/Java/Fundamentals/Concurrency/ThreadTest.java
class ThreadTest{

	private static void handleJob(int id){
		System.out.printf("Thread<%x> has accepted job:%d%n", Thread.currentThread().hashCode(), id);
		Worker.doWork(id);
		System.out.printf("Thread<%x> has finished job:%d%n", Thread.currentThread().hashCode(), id);
	}

	public static void main(String[] args) throws Exception{
		int n = args.length > 0 ? Integer.parseInt(args[0]) : 50;
		Thread child = new Thread(new Runnable(){
			public void run(){
				handleJob(n);
			}
		});
		child.setDaemon(n > 70);
		child.start();
		handleJob(40);
	}
}

